lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -532,38 +557,98 @@ func chanCanBeHopHint(channel *channeldb.OpenChannel, cfg *AddInvoiceConfig) (
 
 // addHopHint creates a hop hint out of the passed channel and channel policy.
 // The new hop hint is appended to the passed slice.
-func addHopHint(hopHints *[]func(*zpay32.Invoice),
-	channel *channeldb.OpenChannel, chanPolicy *channeldb.ChannelEdgePolicy) {
+func addHopHint(hopHints *[][]zpay32.HopHint,
+	channel *HopHintInfo, chanPolicy *channeldb.ChannelEdgePolicy) {
 
 	hopHint := zpay32.HopHint{
-		NodeID:      channel.IdentityPub,
-		ChannelID:   channel.ShortChanID().ToUint64(),
+		NodeID:      channel.RemotePubkey,
+		ChannelID:   channel.ShortChannelID.ToUint64(),
 		FeeBaseMSat: uint32(chanPolicy.FeeBaseMSat),
 		FeeProportionalMillionths: uint32(
 			chanPolicy.FeeProportionalMillionths,
 		),
 		CLTVExpiryDelta: chanPolicy.TimeLockDelta,
 	}
-	*hopHints = append(
-		*hopHints, zpay32.RouteHint([]zpay32.HopHint{hopHint}),
-	)
+
+	*hopHints = append(*hopHints, []zpay32.HopHint{hopHint})
+}
+
+// SelectHopHintsCfg contains the dependencies required to obtain hop hints
+// for an invoice.
+type SelectHopHintsCfg struct {
+	// IsChannelActive is used to generate valid hop hints.
+	IsChannelActive func(chanID lnwire.ChannelID) bool
+
+	// IsPublicNode is returns a bool indicating whether the node with the
+	// given public key is seen as a public node in the graph from the
+	// graph's source node's point of view.
+	IsPublicNode func(pubKey [33]byte) (bool, error)
+
+	// FetchChannelEdgesByID attempts to lookup the two directed edges for
+	// the channel identified by the channel ID.
+	FetchChannelEdgesByID func(chanID uint64) (*channeldb.ChannelEdgeInfo,
+		*channeldb.ChannelEdgePolicy, *channeldb.ChannelEdgePolicy,
+		error)
+}
+
+func newSelectHopHintsCfg(invoicesCfg *AddInvoiceConfig) *SelectHopHintsCfg {
```

Comment:
Do we need to make this function and  `newHopHintInfo` public? In case we want to use `SelectHopHints` from outside LND?

Reply:
I don't think so because `AddInvoiceConfig` is pretty specific to lnd's internals, it has stuff like our `Graph` object and `htlcswitch` in it?